runtime/compiler/env/VMJ9.cpp

```diff
@@ -5970,6 +5970,7 @@ TR_J9VMBase::revertToInterpreted(TR_OpaqueMethodBlock * method)
 int32_t *
 TR_J9VMBase::getStringClassEnableCompressionFieldAddr(TR::Compilation *comp, bool isVettedForAOT)
    {
+   TR_ASSERT(!comp->compileRelocatableCode() || comp->reloRuntime()->isLoading(), "Function cannot be called during AOT method compilation");
```

Comment:
This should be a `TR_ASSERT_FATAL` or we're not going to see this except in a `PROD_WITH_ASSUMES` build (which doesn't yet exist).

Reply:
Addressed in force push.